system/Commands/Generators/ModelGenerator.php

```diff
@@ -101,9 +101,11 @@ protected function prepare(string $class): string
             $baseClass = $match[1];
         }
 
-        $table   = is_string($table) ? $table : plural(strtolower($baseClass));
-        $dbGroup = is_string($dbGroup) ? $dbGroup : 'default';
-        $return  = is_string($return) ? $return : 'array';
+        // Add or remove the DBGroup line based on the presence of the dbgroup option
+        $addDBGroupLine = is_string($dbGroup);
+
```

Comment:
I don't think you need this extra variable. Just pass `$dbGroup` to the template and do the `is_string` check there.

Reply:
Since you changed the behavior, can you add a test that when `--dbgroup <something>` is passed, it still sets the $DBGroup property.

Reply:
> I don't think you need this extra variable. Just pass `$dbGroup` to the template and do the `is_string` check there.

Noted.

Reply:
> Since you changed the behavior, can you add a test that when `--dbgroup <something>` is passed, it still sets the $DBGroup property.

That test is already covered here:
https://github.com/codeigniter4/CodeIgniter4/blob/6f81e6e42e996eb1d0eacb2325600f74fefc9307/tests/system/Commands/ModelGeneratorTest.php#L68

Reply:
Oh I overlooked that.